quattor/functions/validation.pan

```diff
@@ -139,3 +139,15 @@ function valid_absolute_file_paths = {
     };
     true;
 };
+
+@documentation{
+    desc = Function to validate whether the parameter received is a valid username or not; a valid username is considered to only be comprised of alpha numeric characters, plus a few special ones which are: .+:-
+}
+function valid_user_name = {
+    user = to_string(ARGV[0]);
+    if(!match(user, "^[A-Za-z0-9._:-]+$")) {
+        error(user + " is not a valid username!");
+        return(false);
```

Comment:
this will never be reached. if you want the function to be a proper boolean function, remove the `error` (replace by debug)

Reply:
Will the true/false answer be sufficient for using types ?
i.e. `type valid_user = string with valid_user_name;`

Reply:
yes.
do you actually need the function at all? why not make a type to start with, like
```pan
type valid_username = string with match(SELF, someregex);
```
not sure about the error message, if https://github.com/quattor/pan/pull/201 gets merged, the error will contain the failed value

Reply:
I don't mind not having it as a function. I'm just starting doing stuff in pan, so most of what I do might be optimized! Once we find a common ground for the regex itself, I'll probably move to using that directly as you suggest. I suggested to merge the type in https://github.com/quattor/configuration-modules-core/pull/1312

Reply:
since 10.7 you can use a type directly to test a value
before that release, people used to make a separate function if they wanted to test for composite types and such. that's why the current template library has some many type functions

Reply:
The first occurrence of that code was actually a `if (!match(...))` repeated at multiple places, that I replaced by the above function, that then was suggested to me to use to check the type directly instead of having it in the if. Did not catch at that point that I could do the match directly... héhé.

Regarding the type itself, do you know if there is a better place for me to submit it, or the place I did _is_ the best place? The split of all the templates in multiple repositories is quite confusing for now.

Reply:
@jrha any thougths where to put the type? here or in account component?

Reply:
I guess that user names are used for more than just ncm-accounts, so here is ok.